internal/corazawaf/rule.go

```diff
@@ -399,7 +399,7 @@ func (r *Rule) transformArg(arg types.MatchData, argIdx int, cache map[transform
 		default:
 			// NOTE: See comment on transformationKey struct to understand this hacky code
 			argKey := arg.Key()
-			argKeyPtr := (*reflect.StringHeader)(unsafe.Pointer(&argKey)).Data
+			argKeyPtr := (*reflect.StringHeader)(unsafe.Pointer(&argKey)).Data // nolint:staticcheck
```

Comment:
`reflect.StringHeader` has been deprecated and the linter is complaining.
Something like the following might do the trick, but I don't want to rush changes on something very hack and performance-related.
```
argKeyPtr2 := uintptr(unsafe.Pointer(unsafe.StringData(argKey)))
```

Reply:
Ugh, why do we need unsafe in the first place...

Reply:
This is the historical context: https://github.com/corazawaf/coraza/pull/553. We should revisit and see if we have some other viable option. If this looks okay, we can go ahead with merging and I'll open an issue about that

Reply:
Yes, let's upgrade the golang part and create a follow up issue.

Reply:
Filled https://github.com/corazawaf/coraza/issues/1147.